app/controllers/textings_controller.rb

```diff
@@ -0,0 +1,108 @@
+class TextingsController < ApplicationController
+
+  def create
+    # Make a request to Textellent API. If request successful we update our DB
+    response = post_textellent(get_db_data)
+
+    if response['status'] == 'success'
+      recipient = TextingRecipient.find_by(phone_number: texting_params[:phone_number])
+
+      if recipient
+        texting = Texting.new(
+          texting_recipient_id: recipient.id,
+          service_id: texting_params[:service_id]
+          )
+
+        if texting.save
+          render status: :ok, json: { message:'Message sent' }
+        else
+          render status: :bad_request
+        end
+
+      else
+        recipient = TextingRecipient.new(
+          recipient_name: texting_params[:recipient_name],
+          phone_number: texting_params[:phone_number]
+          )
+
+        recipient.save
+
+        texting = Texting.new(
+          texting_recipient_id: recipient.id,
+          service_id: texting_params[:service_id]
+          )
+
+        if texting.save
+          render status: :ok, json: { message:'Message sent' }
+        else
+          render status: :bad_request
+        end
+
+      end
+
+    else
+      render status: :bad_request, json: { error: response['message'] }
+    end
+  end
+
+  private
+
+  # Generate a data object to send to Textellent API
+  def get_db_data
+    recipient_name = texting_params[:recipient_name]
+    phone_number = texting_params[:phone_number]
+    service = Service.find(texting_params[:service_id])
+    category_ids = service.category_ids
+    resource = Resource.find(service.resource_id)
+    address = resource.addresses
+    phone = resource.phones[0].number
+
+    categories = []
+    category_ids.each do |category_id|
+      categories << Category.find(category_id).name
+    end
```

Comment:
Nit: I prefer to use a more functional programming approach of doing this by using the [`#map`](https://ruby-doc.org/core-2.7.0/Array.html#method-i-map) method on arrays. This would look like:

```suggestion
    categories = category_ids.map do |category_id|
      Category.find(category_id).name
    end
```

Also, does `Service` have a `categories` property, rather than the `categories_id` property? You may be able to save some unnecessary database calls by prefetching all the categories and then accessing their names directly. I haven't tested this code, but I think this could look like the following:

```rb
service = Service.includes(:categories).find(texting_params[:service_id])

categories = service.categories.map { |c| c.name }
```

See https://engineering.gusto.com/a-visual-guide-to-using-includes-in-rails/ for a good guide on how `.includes()` and prefetching works in Rails.